### PR TITLE
[DEV-2728] Added capability to do as-of in method lookup in the client

### PR DESCRIPTION
**Pull request by nonibansal**
Added capability to do as-of in method lookup in the client

### CODE REVIEW

File: fennel/client/client.py [hunk 1]
```diff
@@ -705,10 +705,25 @@ def extract_historical_cancel_request(self, request_id):
         )
 
     def lookup(
-        self, dataset_name: str, keys: List[Dict[str, Any]], fields: List[str]
+        self,
+        dataset_name: str,
+        keys: List[Dict[str, Any]],
+        fields: List[str],
+        timestamps: List[Union[int, str]] = None,
```
Comment (aditya-nambiar) [inline fennel/client/client.py:712]: Why is datetime type not there ?

**Review by aditya-nambiar — CHANGES_REQUESTED** [review]
Could you add this to the docs too

File: fennel/client/client.py (hunk 1 above)
Comment (nonibansal) [reply fennel/client/client.py]: added that, forgot about that.

File: docs/pages/api-reference/client.md [hunk 2]
```diff
@@ -175,3 +175,35 @@ The response format of this function and the `extract_historical` function are i
...
+### **lookup**
+
+The `lookup` method allows users to lookup values from keyed datasets.&#x20;
+
+**Arguments:**
+
+* `dataset_name: str`: The name of the dataset against which we want to lookup values.
+* `keys: List[Dict[str, Any]]` : List of dict where each dict can be thought of as one entity for which we want to lookup. 
+* `fields: List[str]`: List of field names in the dataset that we want to lookup.
+* `timestamps: List[Union[int, str, datetime]]` - Pass this argument if we want to lookup as of a specific time. The length of this list should be exactly equal to length of keys argument.
```
Comment (aditya-nambiar) [inline docs/pages/api-reference/client.md:189]: Can we add what the str and ints can be ?

**Review by aditya-nambiar — APPROVED** [review]

File: docs/pages/api-reference/client.md (hunk 2 above)
Comment (nonibansal) [reply docs/pages/api-reference/client.md]: added

**Pull request merged**